pkg/staking/stakingcontract/contract.go

```diff
@@ -0,0 +1,191 @@
+// Copyright 2022 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package stakingcontract
+
+import (
+	"context"
+	"crypto/rand"
+	"errors"
+	"fmt"
+	"math/big"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethersphere/bee/pkg/sctx"
+	"github.com/ethersphere/bee/pkg/transaction"
+	"github.com/ethersphere/go-sw3-abi/sw3abi"
+)
+
+var (
+	MinimumStakeAmount = big.NewInt(1)
+
+	erc20ABI = parseABI(sw3abi.ERC20ABIv0_3_1)
+	//TODO: get ABI for staking contract and replace it below
+	stakingABI = parseABI(sw3abi.ERC20ABIv0_3_1)
+	//TODO: enable below mentioned topic for receiving receipts
+	//stakeUpdatedTopic = stakingABI.Events["StakeUpdated"].ID
+
+	ErrInvalidStakeAmount = errors.New("invalid stake amount")
+	ErrInsufficientFunds  = errors.New("insufficient token balance")
+	ErrNotImplemented     = errors.New("not implemented")
+
+	depositStakeDescription = "Deposit Stake"
+)
+
+type StakingContract interface {
+	DepositStake(ctx context.Context, stakedAmount *big.Int, overlay []byte) error
```

Comment:
1) Using `swarm.Address` would be more indicative instead of using `[]byte` for overlay type

2) Since this method shouldn't mutate `stakedAmount` value, it would be better to pass non-pointer value